stan/math/rev/functor/reduce_sum.hpp

```diff
@@ -18,227 +18,346 @@ namespace stan {
 namespace math {
 namespace internal {
 
+/**
+ * Var specialization of implimentation called by reduce `reduce_sum`.
+ * @tparam ReduceFunction An type with a valid `operator()`
+ * @tparam ReturnType The return type of the call to
+ * `ReduceFunction::operator()`
+ * @tparam Vec type of the first input argument with an `operator[]`
+ * @tparam Args Parameter pack holding the types of the object send to
+ * `ReduceFunction::operator()`
+ */
 template <typename ReduceFunction, typename ReturnType, typename Vec,
           typename... Args>
 struct reduce_sum_impl<ReduceFunction, require_var_t<ReturnType>, ReturnType,
                        Vec, Args...> {
-  template <typename T, require_arithmetic_t<scalar_type_t<T>>...>
-  static inline const T& deep_copy(const T& arg) {
-    return arg;
-  }
+  /**
+   * Internal object used in `tbb::parallel_reduce`
+   * @note see link [here](https://tinyurl.com/vp7xw2t) for requirements.
```

Comment:
I don't think we need to say more than it's for tbb::parallel_reduce. I think I've left links in docs before but that doesn't seem reliable.

Reply:
The url here is a wayback machine link. I think for future maintenance it would be good to know why it's built with an inner struct and has the types it does etc.